percy/lib/percy_automate.py

```diff
@@ -0,0 +1,51 @@
+import os
+import requests
+from appium.webdriver.webdriver import WebDriver
+
+from percy.common import log
+from percy.errors import CLIException
+from percy.errors import DriverNotSupported
+from percy.lib.percy_options import PercyOptions
+from percy.environment import Environment
+
+# Maybe get the CLI API address from the environment
+PERCY_CLI_API = os.environ.get('PERCY_CLI_API') or 'http://localhost:5338'
+IGNORE_ELEMENT_KEY = 'ignore_region_appium_elements'
+
+class PercyOnAutomate:
+    def __init__(self, driver):
+        if not isinstance(driver, WebDriver):
+            raise DriverNotSupported
+        self.driver = driver
+        self.percy_options = PercyOptions(self.driver.capabilities)
+
+    def screenshot(self, name: str, **kwargs):
+        if not self.percy_options.enabled:
+            return None
+        if not isinstance(name, str):
+            raise TypeError('Argument name should be a string')
+
+        try:
+            ignore_region_elements = [element.id for element in kwargs.get(IGNORE_ELEMENT_KEY, [])]
+            kwargs.pop(IGNORE_ELEMENT_KEY, None)
+
+            # Post to automateScreenshot endpoint with driver options and other info
+            response = requests.post(f'{PERCY_CLI_API}/percy/automateScreenshot', json={
```

Comment:
this should be under CLI Wrapper and update similar to how we do for `post_screenshots`